### ISSUE TITLE
Problem in installin it

### ISSUE THREAD

**Issue by Mehul266** [Und3rf10w/kali-anonsurf#88]
I am using kali latest version . i do not know what is the problem. please help
i am pasting the whole code below

```
root@HP:~/kali-anonsurf# ./installer.sh
--2019-05-05 23:08:03--  https://geti2p.net/_static/i2p-debian-repo.key.asc
Resolving geti2p.net (geti2p.net)... 91.143.92.136, 2a02:180:a:65:2456:6542:11
:1010
Connecting to geti2p.net (geti2p.net)|91.143.92.136|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 16431 (16K) [text/plain]
Saving to: ‘/tmp/i2p-debian-repo.key.asc’

/tmp/i2p-debian-rep 100%[===================>]  16.05K  65.6KB/s    in 0.2s   

2019-05-05 23:08:05 (65.6 KB/s) - ‘/tmp/i2p-debian-repo.key.asc’ saved [16431/
431]

OK
Hit:1 http://deb.i2p2.no unstable InRelease    
Reading package lists... Done                  
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package libservlet3.0-java
E: Couldn't find any package by glob 'libservlet3.0-java'
E: Couldn't find any package by regex 'libservlet3.0-java'
--2019-05-05 23:08:07--  http://ftp.us.debian.org/debian/pool/main/j/jetty8/li
etty8-java_8.1.16-4_all.deb
Resolving ftp.us.debian.org (ftp.us.debian.org)... 64.50.233.100, 64.50.236.52
208.80.154.15, ...
Connecting to ftp.us.debian.org (ftp.us.debian.org)|64.50.233.100|:80... conne
ed.
HTTP request sent, awaiting response... 200 OK
Length: 1666404 (1.6M) [application/x-troff-man]
Saving to: ‘libjetty8-java_8.1.16-4_all.deb.1’

libjetty8-java_8.1. 100%[===================>]   1.59M   308KB/s    in 7.4s   

2019-05-05 23:08:15 (220 KB/s) - ‘libjetty8-java_8.1.16-4_all.deb.1’ saved [16
404/1666404]

Selecting previously unselected package libjetty8-java.
(Reading database ... 374608 files and directories currently installed.)
Preparing to unpack libjetty8-java_8.1.16-4_all.deb ...
Unpacking libjetty8-java (8.1.16-4) ...
dpkg: dependency problems prevent configuration of libjetty8-java:
 libjetty8-java depends on libservlet3.0-java; however:
  Package libservlet3.0-java is not installed.

dpkg: error processing package libjetty8-java (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 libjetty8-java
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package libgetopt-java is not available, but is referred to by another package
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package ttf-dejavu is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  fonts-dejavu-extra fonts-dejavu-core fonts-dejavu

Package libservlet3.0-java is not available, but is referred to by another pac
ge.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Unable to locate package libecj-java
E: Package 'libgetopt-java' has no installation candidate
E: Package 'libservlet3.0-java' has no installation candidate
E: Unable to locate package glassfish-javaee
E: Package 'ttf-dejavu' has no installation candidate
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... Done
The following packages will be REMOVED:
  libjetty8-java
0 upgraded, 0 newly installed, 1 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 2,310 kB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 374851 files and directories currently installed.)
Removing libjetty8-java (8.1.16-4) ...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
i2p-keyring is already the newest version (2018.09.16).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package tor is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package secure-delete is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'secure-delete' has no installation candidate
E: Package 'tor' has no installation candidate
dpkg-deb: building package 'kali-anonsurf' in 'kali-anonsurf.deb'.
Selecting previously unselected package kali-anonsurf.
(Reading database ... 374608 files and directories currently installed.)
Preparing to unpack kali-anonsurf.deb ...
Unpacking kali-anonsurf (1.2.2.2) ...
dpkg: dependency problems prevent configuration of kali-anonsurf:
 kali-anonsurf depends on secure-delete; however:
  Package secure-delete is not installed.
 kali-anonsurf depends on tor; however:
  Package tor is not installed.
 kali-anonsurf depends on i2p; however:
  Package i2p is not installed.

dpkg: error processing package kali-anonsurf (--install):
 dependency problems - leaving unconfigured
Processing triggers for systemd (240-4) ...
Errors were encountered while processing:
 kali-anonsurf
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... Done
The following packages will be REMOVED:
  kali-anonsurf
0 upgraded, 0 newly installed, 1 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 65.5 kB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 374618 files and directories currently installed.)
Removing kali-anonsurf (1.2.2.2) ...
Processing triggers for systemd (240-4) ...

Selecting previously unselected package kali-anonsurf.
(Reading database ... 374608 files and directories currently installed.)
Preparing to unpack kali-anonsurf.deb ...
Unpacking kali-anonsurf (1.2.2.2) ...
dpkg: dependency problems prevent configuration of kali-anonsurf:
 kali-anonsurf depends on secure-delete; however:
  Package secure-delete is not installed.
 kali-anonsurf depends on tor; however:
  Package tor is not installed.
 kali-anonsurf depends on i2p; however:
  Package i2p is not installed.

dpkg: error processing package kali-anonsurf (--install):
 dependency problems - leaving unconfigured
Processing triggers for systemd (240-4) ...
Errors were encountered while processing:
 kali-anonsurf

```

**Comment by 3z3kI31** [comment]
I have the same issue!

**Comment by pieterhouwen** [comment]
libservlet3.0-java is no longer in the package list, it has been superseded by 3.1-java:

pieter@SUPERSECRETSERVER:~$ sudo su
[sudo] password for pieter:
root@SUPERSECRETSERVER:/home/pieter# apt search libservlet3.0-java
Sorting... Done
Full Text Search... Done
root@SUPERSECRETSERVER:/home/pieter# apt search libservlet
Sorting... Done
Full Text Search... Done
libservlet-api-java/kali-rolling 4.0.1-2 all
  Java Servlet API

libservlet-api-java-doc/kali-rolling 4.0.1-2 all
  Java Servlet API (documentation)

libservlet3.1-java/kali-rolling 1:4.0.1-2 all
  Java Servlet API 3.1 (transitional package)

root@SUPERSECRETSERVER:/home/pieter#


You can download the old version at: http://security.debian.org/debian-security/pool/updates/main/t/tomcat7/libservlet3.0-java_7.0.56-3+really7.0.94-1_all.deb 

Open the installer.sh script and make the following changes:

On line 17, replace "apt-get install libservlet-3.0-java" with: wget http://security.debian.org/debian-security/pool/updates/main/t/tomcat7/libservlet3.0-java_7.0.56-3+really7.0.94-1_all.deb && dpkg -i libservlet3.0-java_7.0.56-3+really7.0.94-1_all.deb

Save the installer.sh and re-run.

If you're still facing issues please paste your new errors here.

**Comment by azamet90** [comment]
also anonsurf must be updated for the new package huh?
